src/sc/asyncproofverifier.cpp

```diff
@@ -155,319 +95,86 @@ void CScAsyncProofVerifier::RunPeriodicVerification()
+ * from the cswProofs and certProofs maps.
+ * 
+ * 2. All the transactions/certificates that did not pass the verification are rejected
+ * and the sender nodes are notified and the related proofs are removed
+ * from the cswProofs and certProofs maps.
+ * 
+ * 3. All the transactions/certificates that are in an UNKNOWN state are not processed
+ * and are kept into the related maps.
  * 
- * @param cswInputs The map of CSW inputs data to be verified.
- * @param certInputs The map of certificates data to be verified.
- * @return std::pair<bool, std::vector<AsyncProofVerifierOutput>> A pair containing the total result of the whole batch verification (bool) and the list of single results.
+ * @param proofs The set of proofs submitted as input to the proof verifier
  */
-std::pair<bool, std::vector<AsyncProofVerifierOutput>> CScAsyncProofVerifier::BatchVerify(
-    const std::map</*scTxHash*/uint256, std::map</*outputPos*/unsigned int, CCswProofVerifierInput>>& cswInputs,
-    const std::map</*certHash*/uint256, CCertProofVerifierInput>& certInputs) const
+void CScAsyncProofVerifier::ProcessVerificationOutputs(std::map</* Tx hash */ uint256, CProofVerifierItem>& proofs)
 {
-    bool allProofsVerified = true;
-    std::vector<AsyncProofVerifierOutput> outputs;
-
-    CctpErrorCode code;
-    ZendooBatchProofVerifier batchVerifier;
-    uint32_t idx = 0;
-    bool ret = true;
-
-    for (const auto& verifierInput : cswInputs)
+    // Post processing of proofs
+    for (auto i = proofs.begin(); i != proofs.end();)
     {
-        for (const auto& entry : verifierInput.second)
-        {
-            const CCswProofVerifierInput& input = entry.second;
+        CProofVerifierItem item = i->second;
 
-            wrappedFieldPtr sptrScId = CFieldElement(input.scId).GetFieldElement();
-            field_t* scid_fe = sptrScId.get();
- 
-            const uint160& csw_pk_hash = input.pubKeyHash;
-            BufferWithSize bws_csw_pk_hash(csw_pk_hash.begin(), csw_pk_hash.size());
- 
-            wrappedFieldPtr   sptrCdh       = input.certDataHash.GetFieldElement();
-            wrappedFieldPtr   sptrCum       = input.ceasingCumScTxCommTree.GetFieldElement();
-            wrappedFieldPtr   sptrNullifier = input.nullifier.GetFieldElement();
-            wrappedScProofPtr sptrCswProof  = input.cswProof.GetProofPtr();
-            wrappedScVkeyPtr  sptrCeasedVk  = input.ceasedVk.GetVKeyPtr();
-
-            ret = batchVerifier.add_csw_proof(
-                idx,
-                input.nValue,
-                scid_fe,
-                sptrNullifier.get(),
-                &bws_csw_pk_hash,
-                sptrCdh.get(),
-                sptrCum.get(),
-                sptrCswProof.get(),
-                sptrCeasedVk.get(),
-                &code
-            );
-            idx++;
+        if (item.result == ProofVerificationResult::Failed || item.result == ProofVerificationResult::Passed)
```

Comment:
Cosmetic: check if it is Unknow, in that case increment iterator, else do the processing.

Reply:
Fixed.